src/XML/src/FMI3/fmi3_xml_variable.c

```diff
@@ -2160,169 +2205,80 @@ int fmi3_xml_handle_ModelVariables(fmi3_xml_parser_context_t *context, const cha
         fmi3_xml_set_element_handle(context, "Tool",        FMI3_XML_ELM_ID(VariableTool));
     }
     else {
-         /* postprocess variable list */
-
+         // Post-process variable list
         fmi3_xml_model_description_t* md = context->modelDescription;
-        jm_vector(jm_voidp)* varByVR;
-        size_t i, numvar;
-
-        numvar = jm_vector_get_size(jm_named_ptr)(&md->variablesByName);
 
-        /* store the list of vars in original order */
-        {
-            size_t size = jm_vector_get_size(jm_named_ptr)(&md->variablesByName);
-            md->variablesOrigOrder = jm_vector_alloc(jm_voidp)(size,size,md->callbacks);
-            if (!md->variablesOrigOrder) {
-                fmi3_xml_parse_fatal(context, "Could not allocate memory");
-                return -1;
-            }
-            size_t i;
-            for(i= 0; i < size; ++i) {
-                jm_vector_set_item(jm_voidp)(md->variablesOrigOrder, i, jm_vector_get_item(jm_named_ptr)(&md->variablesByName,i).ptr);
+        // Create variablesByName
+        size_t nVars = jm_vector_get_size(jm_voidp)(&md->variablesOrigOrder);
+        jm_vector_resize(jm_named_ptr)(&md->variablesByName, nVars);  // Allocate a lot of memory to avoid resizing
+        jm_vector_resize(jm_named_ptr)(&md->variablesByName, 0);      // Make sure we start pushback at index=0
+        for (size_t i = 0; i < nVars; i++) {
+            jm_named_ptr named;
+            fmi3_xml_variable_t* v = jm_vector_get_item(jm_voidp)(&md->variablesOrigOrder, i);
+            named.name = v->name;
+            named.ptr = v;
+            jm_vector_push_back(jm_named_ptr)(&md->variablesByName, named);
+            if (v->aliases) {
+                fmi3_xml_alias_variable_t* alias;
+                for (size_t j = 0; j < jm_vector_get_size(jm_voidp)(v->aliases); j++) {
+                    alias = jm_vector_get_item(jm_voidp)(v->aliases, j);
+                    named.name = alias->name;
+                    named.ptr = v;
+                    jm_vector_push_back(jm_named_ptr)(&md->variablesByName, named);
+                }
```

Comment:
(I could've just added the aliases to variablesByName after creating the other lists, but it made more sense to me clean it up while at it, i.e. that we add to variablesOrigOrder when we parse, then create the help lists.)